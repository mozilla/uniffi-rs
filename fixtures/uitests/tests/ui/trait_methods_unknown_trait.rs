// Unfortunately, path is relative to a temporary build directory :-/
uniffi_macros::generate_and_include_scaffolding!("../../../../fixtures/uitests/src/trait_methods_unknown_trait.udl");

fn main() { /* empty main required by `trybuild` */}

// We derive all required traits, plus one we don't know about.
#[derive(Debug, PartialEq, Eq, Hash)]
pub struct TraitMethods {}

impl std::fmt::Display for TraitMethods {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "TraitMethods()")
    }
}
