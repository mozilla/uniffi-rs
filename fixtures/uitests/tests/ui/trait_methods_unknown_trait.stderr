error: Failed to generate scaffolding from UDL file at ../../../../fixtures/uitests/src/trait_methods_unknown_trait.udl: Failed to parse UDL: Invalid trait name: PartialEq
 --> tests/ui/trait_methods_unknown_trait.rs:2:1
  |
2 | uniffi_macros::generate_and_include_scaffolding!("../../../../fixtures/uitests/src/trait_methods_unknown_trait.udl");
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `uniffi_macros::generate_and_include_scaffolding` (in Nightly builds, run with -Z macro-backtrace for more info)
